### DISCUSSION TITLE
[Feedback wanted] What HTML tags and attributes are you using in Marp?

### DISCUSSION THREAD

**Discussion by yhatt** [marp-team/marp#501] [General]
When enabling HTML in Marp, what HTML tags and attributes are you using in Marp slides?

## Background

In the next major version of Marp Core, there is a plan to relax the allowlist for raw HTML contents in Markdown.

- marp-team/marp-core#301

Forgetting the config for enabling HTML tags when converting is a frequently mentioned mistake in Marp discussions and issues. To mitigate that, we plan to allow certain HTML tags and attributes, deemed safe from a security perspective, in Markdown without changing settings. [^1]

[^1]: https://github.com/marp-team/marp-core/issues/368#issuecomment-2004426271

Marp team has wanted to examine HTML tags frequently used by Marp users, and to create a default allowlist that is suitable and secure for every Marp user.

But despite a long-standing pinned issue marp-team/marp-core#301, it has not yet received enough feedback to create that.🥺
We welcome any feedback about the HTML tags and attributes you often use in Marp, no matter how minor!

**Comment by obiwac** [comment]
`kbd` for keyboard shortcuts is one i use quite often :)

**Comment by codingluke** [comment]
Mostly simple `<div>` with classes for colums but also `<script type="module">` which for sure will not be allowed 😅

**Comment by manics** [comment]
Based on a quick `git grep` of my presentations markdown:
`br`
`div`
`img`
`p`
`span`

**Comment by Chiogros** [comment]
`<link>` and `<script>` since I didn't find a way to use `highlight.js` without including its JS (using VSCode extension). It's my only use case.

`<sup>`
`<br>`

**Comment by yhatt** [comment]
Thanks for many feedbacks! :sparkles:

A new Marp Core v4 has been relaxed allowlist of HTML elements and attributes, and a lot of HTML elements for layouting are permited by default. Check out #533 for more details.
